---
title: "Fig_2B"
author: "Max Schubert"
date: "6/18/2019"
output: html_document
editor_options: 
  chunk_output_type: console
params:
  encoding: UTF-8
  warning: FALSE
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE)
pdf.options(encoding = "CP1250")

library(data.table)
library(stringr)
library(plyr)
require(ggplot2)
require(data.table)
require(scales)
library(cowplot)
library(grid)
library(gridExtra)
library(here)
library(ggpubr)

library(ggbeeswarm) #first install.packages("ggbeeswarm")


parent_folder <- here::here()
source(here::here('constants.R'))

```

# fig1B, beta editing across genotypes

```{r fig2_load_data_and_metadata}
fig_2B_dir = "fig_2B"
current_folder <- paste(parent_folder, fig_2B_dir, sep = '/')

#read in data
edit_data_file = paste(current_folder,"b18b19b23_edit_mergedbins", sep = "/")
edit_data_dt <- data.table(fread(edit_data_file))
edit_data_melt <- melt(edit_data_dt, id.var=c('read_id', 'total_count', 'bc_read'), value.name='Count', variable.name='Sample')
#make new col Sample_num for merging to manifest
edit_data_melt[, Sample_num := as.numeric(str_replace_all(Sample, "_cutcount", ""))]

#import manifest
#this contains the data "expt_set" and "tx" for annotating metadata
manifest_file = paste(current_folder, "FC_04592_1234_PF_mm1_mschubert_manifest_edited.csv", sep = "/")
manifest_dt <- data.table(fread(manifest_file)) 
manifest_dt = Filter(function(x)!all(is.na(x)), manifest_dt) #kills columns with all NA
manifest_dt = manifest_dt[,c('Sample_num','expt_set','well_indexed','tx','editing_timepoint', 'control_status')] #keeps only specified columns

#merge manifest to edit_data
melt_with_annot <- edit_data_melt[manifest_dt, on = "Sample_num", nomatch = 0]

#import metadata, merge 
tx_deets_file = paste(current_folder, 'edit_tx_map.csv', sep = "/")
tx_deets = data.table(fread(tx_deets_file)) 
melt_with_metadata <- melt_with_annot[tx_deets, on = c("expt_set","tx"), nomatch = 0]

#import ref_alt, merge
ref_alt_file = paste(current_folder, 'ref_alt_map.csv', sep = "/")
ref_alt = data.table(fread(ref_alt_file)) 
melt_with_metadata2 <- melt_with_metadata[ref_alt, on = "adapters", nomatch = 0]

```

```{r fig1be_ref_alt}
#establishes control sample and normalizes to this, prints into "ratio", kill singletons
melt_with_metadata2[, is_ref := (bc_read == ref)]
melt_with_metadata2[, is_alt := (bc_read == alt)]

melt_with_fedit <- melt_with_metadata2[ is_ref ==TRUE | is_alt == TRUE , fraction_edited := 1- (Count/sum(Count)) , by='Sample_num'][is_ref == TRUE]

```
```{r keep track of reads}
total_counts_merged = sum(melt_with_metadata2$Count)
total_counts_ref_or_alt = sum(melt_with_metadata2[is_ref == TRUE | is_alt == TRUE]$Count)

#compare ref_alt read num to total merged
print(paste(as.character(total_counts_ref_or_alt/total_counts_merged*100), "% of reads are ref or alt seqs", sep = " "))

```

### append data from "H7" experiment in fig S3A showing results for cspRecT, for inclusion here
```{r get_h7_data}
h7_scrubbed = readRDS(file = "h7data")
h7_triple <- h7_scrubbed[genotype == "sbcB-, recJ-, mutS-" ]
h7_triple$genotype = "∆mutS, ∆recJ, ∆sbcB, + CspRecT"

#also scrub fig2B data (beta as SSAP) for plotting
b22 <- melt_with_fedit[expt_set == 'b22']
b22 <- b22[control_status != 'background']
b22_scrubbed <- copy(b22)
b22_scrubbed[, well := well_indexed]
b22_scrubbed[, ssap := "Beta"]
b22_scrubbed[, genotype := strain_genotype]
b22_scrubbed[, locus := adapters]
b22_scrubbed[fraction_edited != NaN]
b22_scrubbed = b22_scrubbed[, c("read_id", "total_count", "bc_read", "Sample", "Count",
                       "Sample_num", "expt_set", "well_indexed", "tx", "editing_timepoint",
                       "control_status", "parent", "strain_genotype", "plas", "adapters", "comment",
                       "plas_desc", "ref", "alt", "is_ref", "is_alt") := NULL]

genotype_rbind = rbind(h7_triple, b22_scrubbed)
```

#explore and output raw data and summary stats
```{r output_and_summarize}
#export raw data, for supplemental table
setorder(genotype_rbind, col = genotype)
output_to = paste(parent_folder, 'supplemental_materials','supplemental_table_1_raw_out.csv', sep = "/")
fwrite(genotype_rbind,output_to)


#aggregated data
genotype_rbind = na.omit(genotype_rbind, cols="fraction_edited")
genotype_rbind_means <- genotype_rbind[,
                        list(
                          mean_fraction_edited = mean(fraction_edited),
                          sd_fraction_edited = sd(fraction_edited),
                          replicates = .N,
                          SEM_fraction_edited = sd(fraction_edited)/sqrt(.N)
                          ), by=c("locus", "genotype")]

#export aggregated data, for supplemental table
setorder(genotype_rbind_means, col = mean_fraction_edited)
output_to = paste(parent_folder, 'supplemental_materials','supplemental_table_1_out.csv', sep = "/")
fwrite(genotype_rbind_means,output_to)

```

#optional, example calculations for numbers used in figure 2 text
```{r calculations_for_fig2_text}
### examine the improvement of deltamutS over WT
genotype_rbind_means[(genotype == "∆mutS") & (locus == "gyrA"), mean_fraction_edited] / genotype_rbind_means[(genotype == "WT") & (locus == "gyrA"), mean_fraction_edited]
genotype_rbind_means[(genotype == "∆mutS") & (locus == "rpoB"), mean_fraction_edited] / genotype_rbind_means[(genotype == "WT") & (locus == "rpoB"), mean_fraction_edited]

### examine improvement of dual exo KO over solely deltamutS
genotype_rbind_means[(genotype == "∆mutS, ∆recJ, ∆sbcB") & (locus == "gyrA"), mean_fraction_edited] / genotype_rbind_means[(genotype == "∆mutS") & (locus == "gyrA"), mean_fraction_edited]
genotype_rbind_means[(genotype == "∆mutS, ∆recJ, ∆sbcB") & (locus == "rpoB"), mean_fraction_edited] / genotype_rbind_means[(genotype == "∆mutS") & (locus == "rpoB"), mean_fraction_edited]

### examine improvement of cspRecT over Beta
genotype_rbind_means[(genotype == "∆mutS, ∆recJ, ∆sbcB, + CspRecT") & (locus == "gyrA"), mean_fraction_edited] / genotype_rbind_means[(genotype == "∆mutS, ∆recJ, ∆sbcB") & (locus == "gyrA"), mean_fraction_edited]
genotype_rbind_means[(genotype == "∆mutS, ∆recJ, ∆sbcB, + CspRecT") & (locus == "rpoB"), mean_fraction_edited] / genotype_rbind_means[(genotype == "∆mutS, ∆recJ, ∆sbcB") & (locus == "rpoB"), mean_fraction_edited]
```

## Fig 2B (with csprecT results)
###setup
```{r fig_2B_setup}
#to make Xlabels have multiple lines, instead of comma and space:
addline_format <- function(x,...){
  gsub(', ','\n',x)
}

#change plot order of genotypes, specify statistical comparisons
genotype_rbind$genotype <- factor(genotype_rbind$genotype,
                                  levels = c("WT","∆mutS",
                                             "∆mutS, ∆sbcB",
                                             "∆mutS, ∆recJ",
                                             "∆mutS, ∆recJ, ∆sbcB",
                                             "∆mutS, ∆recJ, ∆sbcB, + CspRecT"))

my_comparisons <- list( c("WT", "∆mutS"),
                        c("∆mutS", "∆mutS, ∆sbcB"),
                        c("∆mutS","∆mutS, ∆recJ"), #exos independently
                        c("∆mutS, ∆recJ", "∆mutS, ∆recJ, ∆sbcB"), #exos together
                        c("∆mutS, ∆recJ, ∆sbcB", "∆mutS, ∆recJ, ∆sbcB, + CspRecT"))

x_labels <- addline_format(c("WT", "∆mutS", "∆mutS, ∆sbcB", "∆mutS, ∆recJ", "∆mutS, ∆recJ, ∆sbcB", "∆mutS, ∆recJ, ∆sbcB, + CspRecT"))

scientific_10 <- function(x) {
  parse(text=gsub("e", " %*% 10^", scales::scientific_format()(x)))
}

y_breaks <- c(.00001, .0001, .001, .01, .1, .5, 1)

```

###compile figure
```{r fig_2B_plot, fig.height=3, fig.width=4, dpi=200}

fig2B <- ggplot(genotype_rbind, aes(x=genotype, y=fraction_edited, color=locus)) + 
  geom_jitter(width = 0.3, #height = 0.1,
              size = 3, alpha = 0.7, stroke = 0) +
  #geom_quasirandom(method='pseudorandom', size = 3, alpha = 0.7, stroke = 0)+
  scale_y_continuous(breaks = y_breaks,
                     label=c(scientific_10(y_breaks[1:3]), y_breaks[4:7]), 
                     trans = log10_trans()) +
  labs(
    y = "Fraction Edited",
    color = "Locus") +
  stat_compare_means(comparisons = my_comparisons, method = "t.test", label = "p.value", label.y = c(-1.5, -1, -.5, 0, 0.3)) +
  scale_x_discrete(labels = addline_format(x_labels)) +
  theme_linedraw() +
  theme(
    text = element_text(size=fig_font_size),
    axis.text = element_text(size = fig_font_size),
    axis.title = element_text(size = fig_title_font_size),
    axis.title.x = element_blank(),
    strip.background=element_blank(), strip.placement='outside',
    strip.text=element_text(color='black'), 
    #axis.title.x=element_text(margin = margin(1.2*fig_font_size,0,0,0, "pt")), 
    panel.border=element_blank(),
    axis.line=element_line(),
    panel.grid.major.x=element_blank(),
    panel.grid.minor.x=element_blank(),
    panel.grid.minor.y=element_blank(),
    legend.position = c(0.18, 0.95),
    legend.title = element_blank(),
    legend.box = "horizontal",
    legend.direction = "horizontal")

fig2B
```

```{r save fig_2B}
fig_width=5.5
fig_height = 3.33

ggsave(here::here('fig_2B_out.pdf'), plot=fig2B,
       width=fig_width, height=fig_height,
       units='in', dpi=300, device = cairo_pdf)

```






###optional: compute the differences between alleles, across genotypes
### in general, as efficiency improves, differences between alleles decrease
```{r locus_diff}

absolute_ratio <- function(x,y){
  val1 = abs(log10(x)-log10(y))
  return(10^(val1))
}

genotype_rbind_locus_ratios <- genotype_rbind_means[,
                        list(
                          locus_abs_ratio = absolute_ratio(
                            mean_fraction_edited[locus == "gyrA"],
                            mean_fraction_edited[locus == "rpoB"])
                          ), by=c("genotype")]

genotype_rbind_locus_ratios$genotype <- factor(genotype_rbind_locus_ratios$genotype,
                                  levels = c("WT","∆mutS",
                                             "∆mutS, ∆sbcB",
                                             "∆mutS, ∆recJ",
                                             "∆mutS, ∆recJ, ∆sbcB",
                                             "∆mutS, ∆recJ, ∆sbcB, + CspRecT"))


x_labels <- addline_format(c("WT", "∆mutS", "∆mutS, ∆sbcB", "∆mutS, ∆recJ", "∆mutS, ∆recJ, ∆sbcB", "∆mutS, ∆recJ, ∆sbcB, + CspRecT"))
figdiff <- ggplot(genotype_rbind_locus_ratios,
                  aes(x=genotype, y=locus_abs_ratio)) + 
  geom_point(size = 2) +
  labs(
    y = "Editing difference between loci, ratio",
    x = "Genotype") +
  scale_x_discrete(labels = x_labels) +
  theme_linedraw() +
  theme(
    text = element_text(size=fig_font_size),
    axis.text = element_text(size = fig_font_size),
    axis.title = element_text(size = fig_title_font_size),
    axis.title.x = element_blank(),
    panel.border=element_blank(),
    axis.line=element_line(),
    panel.grid.major.x=element_blank(),
    panel.grid.minor.x=element_blank(),
    panel.grid.minor.y=element_blank(),
)

figdiff

###removing individual exo KO data, as recJ KO seems to benefit one over the other
x_labels <- addline_format(c("WT", "∆mutS", "∆mutS, ∆recJ, ∆sbcB", "∆mutS, ∆recJ, ∆sbcB, + CspRecT"))
figdiff2 <- ggplot(genotype_rbind_locus_ratios[(genotype != "∆mutS, ∆sbcB") & (genotype != "∆mutS, ∆recJ")],
                  aes(x=genotype, y=locus_abs_ratio,group = genotype)) + 
  geom_bar(stat = "identity", size = 2) +
  #geom_quasirandom(method='pseudorandom', size = 3, alpha = 0.7, stroke = 0)+
  scale_y_continuous(breaks = c(0,1,2,3,4,5,6,7,8,9)) +
  labs(
    y = "Editing difference between loci, ratio",
    x = "Genotype") +
  scale_x_discrete(labels = x_labels) +
  theme_linedraw() +
  theme(
    text = element_text(size=fig_font_size),
    axis.text = element_text(size = fig_font_size),
    axis.title = element_text(size = fig_title_font_size),
    axis.title.x = element_blank(),
    panel.border=element_blank(),
    axis.line=element_line(),
    panel.grid.major.x=element_blank(),
    panel.grid.minor.x=element_blank(),
    panel.grid.minor.y=element_blank(),
)

figdiff2

```

