---
title: "Fig_4CDE"
author: "Max Schubert"
date: "6/18/2019"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(data.table)
#library(stringr)
library(plyr)
library(gridExtra)
require(ggplot2)
require(data.table)
require(scales)
library(ggpubr)
library(plotly)
#dan's suggestion for plotting:
library(ggrepel)

parent_folder = here::here('fig_4b')
source(here::here('constants.R'))

#set global variables for shifting plotting:
ori_pos = 3900000
ter_pos = 1700000
shift_over_value = 1e6
genome_size = 4641652

```

#background
### this communicates that the whole genome is covered, and that coverage is decently even
```{r before_coverage_data}
#s2 data as 4C
samplenum = 's2'
depthspath <- "/Users/Max_Schubert/dropbox_hms/active_seq_reads/NovNextseq_again/FC_04398/Unaligned_1234_PF_mm1/Data/bwa_sam_bam_depths/"
path_cov <- paste(depthspath,samplenum, "_depths", sep = "")
cov <- fread(path_cov)
names(cov) <- c('chr','pos','count')

#slice into windows, label window as middle base of window
window_size <- 1000
cov[, window := cut(pos,
                    breaks=c(seq(0, max(pos), window_size),
                             max(pos)),
                    labels=c(seq(window_size/2, max(pos)-window_size/2, window_size),
                             max(pos)-window_size/2))]

#calculate average depth per window
wcov_new <- cov[, list(mean_window_count=mean(count)), by=window]

#change window name to genomic position, number
wcov_new[, window := as.numeric(window)*window_size]

```

```{r before_coverage_plot}

#plotting absolute, dots
before_cov_plot_dots <- ggplot(wcov_new) + 
  theme_bw() +
  labs( y = "Coverage Depth\n(Unique sequences)",
        x = "Genome Position") +
  theme(
    strip.background=element_blank(), #strip.placement='outside',
    strip.text=element_text(color='black'), 
    #axis.title.x = element_blank(), 
    panel.border=element_blank(),
    axis.line=element_line(),
    legend.position = 'none') +
  #label ori and ter for clarity:
  geom_vline(xintercept = ori_pos,
             linetype="dashed", color = "black",
             alpha = 0.2, size = 1) +
  geom_vline(xintercept = ter_pos,
             linetype="dashed", color = "black",
             alpha = 0.2, size = 1) +
  annotate("text",
           x = ori_pos-200000, y = 175, label = "Ori") + #annotate ori, adjusted
  annotate("text",
           x = ter_pos-200000, y = 175, label = "Ter") +
  #actual window coverage data as points:
  geom_point(aes(y=mean_window_count, x=window,
                 alpha = 0.001, stroke = 0)) +
  #line describing mean coverage
  geom_hline(yintercept = mean(wcov_new$mean_window_count),
             linetype="dashed", color = "red", size = 1) 
 #+

before_cov_plot_dots

ggsave('fig_4C_coverage.pdf', plot=before_cov_plot_dots, width=5, height=2, units='in', dpi=300, useDingbats = FALSE)
```

# data for enrichment of alleles, 
```{r get_enriched_SNP_data}
SNP_table = "a3a4.csv"

SNP_path = paste(parent_folder, SNP_table, sep = "/")
SNP_file_dt = data.table(fread(SNP_path))

#ditch columns with na's
#cutting down table to only a few columns
SNP_file_dt = Filter(function(x)!all(is.na(x)), SNP_file_dt) #kills columns with all NA
SNP_file_trimmed_dt = SNP_file_dt[,c( "UID", "POSITION",
                                        "REF", "ALT",
                                        "EXPERIMENT_SAMPLE_LABEL", "DP",
                                        "INFO_EFF_CONTEXT","INFO_EFF_EFFECT",
                                        "INFO_EFF_AA","INFO_EFF_GT",
                                        "INFO_EFF_CLASS", "INFO_EFF_GENE")] #keeps only specified columns

```


```{r calc_relative_depth_of_snps}
#calculating SNP depth relative to maximum
SNP_file_trimmed_dt[, dp_rel_max := DP/max(DP), by='EXPERIMENT_SAMPLE_LABEL']
#average replicates into new column
SNP_file_trimmed_dt[, mean_dp_rel_max := mean(dp_rel_max), by = UID]

#calculating coverage depth relative to maximum
wcov_new[, dp_rel_max := mean_window_count/max(mean_window_count)]

```

### plot "before" and "after_SNPs" in polar coordinates
```{r plot_all_polar}
all_polar_plot <- ggplot() +

  #data for "before", offset by one to create a circle
  geom_freqpoly(data = wcov_new, stat = "identity",
              aes(y= dp_rel_max + 1 , x=window, color = "Library coverage, before treatment"), color='black', alpha = 0.4) +
  
  #data for "after", offset by one to create a circle
  geom_freqpoly(data = SNP_file_trimmed_dt,
                  aes(x=POSITION, y = mean_dp_rel_max + 1),
                  stat = "identity", size = 1, color = "#E69F00") +
    
    theme(text = element_text(size=18)) +
    theme_minimal() +
    labs( y = "Coverage Depth (relative)",
          x = "Genomic Position (Mbp)") +
  scale_y_continuous(limits = c(0,2)) +
  scale_x_continuous(breaks = c(0,1E6, 2E6, 3E6, 4E6), #issues with 0
                     labels = c("0", "1", "2", "3", "4")) +
    theme(
      #strip.background=element_blank(), strip.placement='outside',
      strip.text=element_text(color='black'), 
      axis.title.x=element_text(margin = margin(1.2*fig_font_size,0,0,0, "pt")), 
      panel.border=element_blank(),
      panel.grid.minor.x=element_blank(),
      panel.grid.minor.y=element_blank(),
      axis.line=element_line(), #+#+#,
      legend.position = 'none') +
    coord_polar()

all_polar_plot

ggsave('fig_4D.pdf', plot=all_polar_plot, width=4, height=4, units='in', dpi=300, useDingbats = FALSE)
```

```{r }

```





#########################3
#cruft below:

###these are the hits, ori and ter annotated with O and X
```{r plot_relative_depth_of_snps}

# this is still in progress...
after_coverage_plot <- ggplot(test_file_trimmed_dt, aes(x=POSITION, y=dp_rel_sum, color=EXPERIMENT_SAMPLE_LABEL)) + 
    geom_freqpoly(stat = "identity", size = 1, alpha = 0.5) + #position = position_dodge(width = 2),
    geom_jitter(size = 1, alpha = 0.5) +
    theme_bw() +
    theme(text = element_text(size=18)) +
    facet_wrap(EXPERIMENT_SAMPLE_LABEL~., ncol=1) +
    theme_minimal() +
    theme(text = element_text(size=fig_font_size)) +
    labs( y = "Coverage Depth (Percent of SNPs observed)",
          x = "Genomic Position (bp)") +
    scale_y_continuous(breaks = c(0, 10, 20, 30, 40, 50)) +
    theme(
      strip.background=element_blank(), strip.placement='outside',
      strip.text=element_text(color='black'), 
      axis.title.x=element_text(margin = margin(1.2*fig_font_size,0,0,0, "pt")), 
      panel.border=element_blank(),
      panel.grid.minor.x=element_blank(),
      panel.grid.minor.y=element_blank(),
      axis.line=element_line(),
      legend.position = 'none') +
    annotate("text", x = 3900000, y = -.01, label = "O") + #annotate ori
    annotate("text", x = 1700000, y = -.01, label = "X") #annotate ter

#shifted version with modulus
after_shifted = test_file_trimmed_dt[, position_modulus := (POSITION+shift_over_value)%%(genome_size - shift_over_value)]

after_coverage_plot_shifted <- ggplot(test_file_trimmed_dt, aes(x=position_modulus, y=dp_rel_sum, color=EXPERIMENT_SAMPLE_LABEL)) + 
    #geom_freqpoly(stat = "identity", size = 1, alpha = 0.5) + #position = position_dodge(width = 2),
    geom_col(position = position_dodge(width = 2), size = 1, alpha = 0.5) + #
    geom_jitter(size = 1, alpha = 0.5) +
    theme_bw() +
    theme(text = element_text(size=18)) +
    facet_wrap(EXPERIMENT_SAMPLE_LABEL~., ncol=1) +
    theme_minimal() +
    theme(text = element_text(size=fig_font_size)) +
    labs( y = "Coverage Depth (Percent of SNPs observed)",
          x = "Genomic Position (bp)") +
    scale_y_continuous(breaks = c(0, 10, 20, 30, 40, 50)) +
    theme(
      strip.background=element_blank(), strip.placement='outside',
      strip.text=element_text(color='black'), 
      axis.title.x=element_text(margin = margin(1.2*fig_font_size,0,0,0, "pt")), 
      panel.border=element_blank(),
      panel.grid.minor.x=element_blank(),
      panel.grid.minor.y=element_blank(),
      axis.line=element_line(),
      legend.position = 'none') +
    annotate("text", x = (3900000+shift_over_value)%%(genome_size - shift_over_value), y = -.01, label = "O") + #annotate ori
    annotate("text", x = (1700000+shift_over_value)%%(genome_size - shift_over_value), y = -.01, label = "X") #annotate ter

#plot both in grid
grid.arrange(
  grobs = list(after_coverage_plot, after_coverage_plot_shifted),
  #widths = c(2, 1, 1),
  layout_matrix = rbind(c(1, 1),
                        c(2, 2))
)


#after_coverage_plot
```

#test version with relative position
```{r calc_relative_depth_of_snps}
test_file_trimmed_dt[, dp_rel_sum := DP/sum(DP), by='EXPERIMENT_SAMPLE_LABEL']
test_file_trimmed_dt[, dp_rel_mean := DP/mean(DP), by='EXPERIMENT_SAMPLE_LABEL']
test_file_trimmed_dt[, dp_rel_median := DP/median(DP), by='EXPERIMENT_SAMPLE_LABEL']

#test_file_trimmed_dt[, rel_enrichment_dp := DP/mean(DP), by='EXPERIMENT_SAMPLE_LABEL']
#test_file_trimmed_dt[, dp_rel_mean_log := log(dp_rel_mean)]
#add a differing "bump" so that logged values are in different positions on the x
#test_file_trimmed_dt_with_offset = test_file_trimmed_dt[ EXPERIMENT_SAMPLE_LABEL == "291_303", offset := 29]
#test_file_trimmed_dt_with_offset[ EXPERIMENT_SAMPLE_LABEL == "292_304", offset := 30]

#test_file_trimmed_dt_with_offset[, rel_enrichment_dp_log_offset := dp_rel_mean_log + offset]
#test_file_trimmed_dt_with_offset[, dp_rel_mean_offset := dp_rel_mean + offset]

```


#idea one:
### plot in normal XY, with coverage before on same plot
```{r plot_rel_depths}

# this is still in progress...
after_coverage_plot <- ggplot() +
    #data for "after"
    geom_bar(data = test_file_trimmed_dt,
                  aes(x=POSITION, y=dp_rel_median, color=EXPERIMENT_SAMPLE_LABEL),
                  stat = "identity", size = 1, alpha = 1) +
    #geom_point(size = 2) +
  
  #data for "before"
    #geom_freqpoly(data = wcov_new, stat = "identity",
    #          aes(y=dp_rel_mean, x=window, color = "gray30")) +
    geom_freqpoly(data = wcov_new, stat = "identity",
              aes(y=dp_rel_median, x=window, color = "gray30")) +
    
    theme(text = element_text(size=18)) +
    theme_minimal() +
    labs( y = "Coverage Depth (relative)",
          x = "Genomic Position (Mbp)") +
    scale_y_continuous(#breaks = c(20, 30, 40, 50, 60, 70, 80), 
                       #labels = c(0, 10, 20, 30, 40, 50, 60), 
                       trans = log10_trans()) + #, #imposing offset of 20, because 20 was added to data to make the chart hollow
                       #limits = c(0, 77)) +
    scale_x_continuous(breaks = c(1E6, 2E6, 3E6, 4E6, genome_size), #having issues marking zero
                       labels = c("1", "2", "3", "4", "0")) +
    theme(
      strip.background=element_blank(), strip.placement='outside',
      strip.text=element_text(color='black'), 
      axis.title.x=element_text(margin = margin(1.2*fig_font_size,0,0,0, "pt")), 
      panel.border=element_blank(),
      panel.grid.minor.x=element_blank(),
      panel.grid.minor.y=element_blank(),
      axis.line=element_line()) #+#,
      #legend.position = 'none') +
    #coord_polar()

after_coverage_plot
```



```{r plot_both}


```



```{r zoom_plot}
#for zoom in on folA locus
zoomed_dt = test_file_trimmed_dt[POSITION < 60000]
zoomed_dt = subset(zoomed_dt, EXPERIMENT_SAMPLE_LABEL = "291_303")


ggplot(zoomed_dt, aes(x=POSITION, y=DP)) + 
  geom_col(position = position_dodge(width = 0.9), fill = "#FF6666", size =2 ) +
  theme_bw() +
  scale_y_continuous(trans = log10_trans()) +
  theme(text = element_text(size=24)) +
  theme_bw() +
  labs(
    x = "Mutations, shown at their position on the chromosome",
    y = "Depth of coverage",
    title = "Loci conferring TMP resistance in sheared gDNA RLR")

#optional, to inspect plot
#ggplotly()


#version that is less zoomed
zoomed_dt = test_file_trimmed_dt[POSITION < 1000000]
zoomed_dt = subset(zoomed_dt, EXPERIMENT_SAMPLE_LABEL = "291_303")


ggplot(zoomed_dt, aes(x=POSITION, y=DP)) + 
  geom_col(position = position_dodge(width = 0.9), fill = "#FF6666", size = 100) +
  theme_bw() +
  scale_y_continuous(trans = log10_trans()) +
  theme(text = element_text(size=24)) +
  theme_bw() +
  labs(
    x = "Mutations, shown at their position on the chromosome",
    y = "Depth of coverage",
    title = "Loci conferring TMP resistance in sheared gDNA RLR")

```


#test version with shift by modulus of X
```{r plot_relative_depth_of_snps}
test_file_trimmed_dt[, rel_dp := DP/sum(DP), by='EXPERIMENT_SAMPLE_LABEL']
test_file_trimmed_dt[, rel_enrichment_dp := DP/mean(DP), by='EXPERIMENT_SAMPLE_LABEL']

#shift plot, so that rep origin appears near plot origin
test_file_trimmed_dt[, position_modulus := (POSITION+shift_over_value)%%(genome_size - shift_over_value)]

# this is still in progress...
after_coverage_plot <- ggplot(test_file_trimmed_dt, aes(x=position_modulus, y=rel__enrichment_dp, color=EXPERIMENT_SAMPLE_LABEL)) + 
    geom_freqpoly(stat = "identity", size = 1, alpha = 1) + #position = position_dodge(width = 2),
    geom_jitter(size = 1) +
    theme_bw() +
    theme(text = element_text(size=18)) +
    facet_wrap(EXPERIMENT_SAMPLE_LABEL~., ncol=1) +
    theme_minimal() +
    theme(text = element_text(size=fig_font_size)) +
    labs( y = "Coverage Depth (Percent of SNPs observed)",
          x = "Genomic Position (bp) MODULUS") +
    theme(
      strip.background=element_blank(), strip.placement='outside',
      strip.text=element_text(color='black'), 
      axis.title.x=element_text(margin = margin(1.2*fig_font_size,0,0,0, "pt")), 
      panel.border=element_blank(),
      panel.grid.minor.x=element_blank(),
      panel.grid.minor.y=element_blank(),
      axis.line=element_line(),
      legend.position = 'none') +
    annotate("text",
             x = (ori_pos+shift_over_value)%%(genome_size - shift_over_value),
             y = -.01, label = "O") + #annotate ori, adjusted
    annotate("text",
             x = (ter_pos+shift_over_value)%%(genome_size - shift_over_value),
             y = -.01, label = "X") #annotate ter, adjusted


after_coverage_plot
```
