---
title: "Fig_2D"
author: "Max Schubert"
date: "8/21/2019"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(data.table)
library(stringr)
library(plyr)
library(gridExtra)
require(ggplot2)
require(data.table)
require(scales)
library(ggpubr)
library(plotly)
#library(dplyr)

library(ggbeeswarm) #first install.packages("ggbeeswarm")


parent_folder <- here::here('fig_2D')

source(here::here('constants.R'))
```

## reanalysis of experiment "H10"

```{r get_data}
#different places than last time
data_dir = parent_folder
analysis_dir = parent_folder

#data
edit_data_name = "h10_merge_bins_out"
edit_data_file = paste(data_dir, edit_data_name, sep = "/")
edit_data_dt <- data.table(fread(edit_data_file))
edit_data_melt <- melt(edit_data_dt, id.var=c('read_id', 'total_count', 'bc_read'), value.name='Count', variable.name='Sample')
#use Sample_ID to merge to metadata
edit_data_melt[, well := str_replace_all(Sample, "_cutcount", "")]
edit_data_melt[, col := as.integer(gsub("[^0-9]", "", well))]
```

```{r read_in_metadata}
#manifest, describing the sample sequenced
manifest_name = "col2adapters.csv"
manifest_file = paste(analysis_dir, manifest_name, sep = "/")
manifest_dt <- data.table(fread(manifest_file))
#make smaller version to use as metadata
metadata = manifest_dt[, c("adaptrset",
                "cutadapt_edit_5prime", "cutadapt_edit_3prime",
                "cutadapt_5prime", "cutadapt_3prime"):=NULL]


#merge manifest to edit_data (merge changed from Sample_ID to 'col')
melt_with_annot <- edit_data_melt[metadata, on = "col", nomatch = 0]
melt_with_annot_nozero = melt_with_annot[Count > 0] #kill zero values to make table smaller
```

# measure fraction edited

```{r fraction_edited}

melt_with_annot_nozero[, is_ref := (bc_read == ref)]
melt_with_annot_nozero[, is_alt := (bc_read == alt)]

#of sequences matching ref or alt, what fraction is edited
melt_with_fedit <- melt_with_annot_nozero[ is_ref ==TRUE | is_alt == TRUE , fraction_edited := 1- (Count/sum(Count)) , by='well'][is_ref == TRUE]
melt_with_fedit[, c("is_ref", "is_alt"):=NULL] #ditch these columns now

#save this table
#save_to = 'h10fraction_edited.rds'
#saveRDS(melt_with_fedit, file = paste(analysis_dir,save_to, sep = '/'))

```
```{r keep track of reads}

total_counts_merged = sum(melt_with_annot_nozero$Count)
total_counts_ref_or_alt = sum(melt_with_annot_nozero[is_ref == TRUE | is_alt == TRUE]$Count)


#compare ref_alt read num to total merged
print(paste(as.character(total_counts_ref_or_alt/total_counts_merged*100), "% of reads are ref or alt seqs", sep = " "))

```


### append data from "H7" experiment in supp fig S3A showing results for cspRecT, for inclusion here. also plotted in 2B
```{r get_h7_data}
h7_scrubbed = readRDS(file = "h7data")
h7_triple <- h7_scrubbed[genotype == "sbcB-, recJ-, mutS-" ]
h7_triple$genotype = "∆mutS, ∆recJ, ∆sbcB, + CspRecT"
h7_triple$label = ""

#combine data
h10 <- melt_with_fedit
h10$label = "TAG > TAA replacement, by gene"
fig2d_rbind = rbind(h7_triple, h10, fill = TRUE)

#summarize data, for visualizing mean efficiency across loci
fig2d_mean_editing_by_locus <- fig2d_rbind[,
                        list(
                          mean_fraction_edited = mean(fraction_edited)
                          ), by=c("locus")]
#export these means, for use in supplemental figure 5B
savesummto = paste(here::here(),"supplemental_figures","fig_s5","2D_summ.rds", sep = "/")
saveRDS(fig2d_mean_editing_by_locus, file = savesummto)
```


#plotting for figure 2D
```{r set_up_plot}
mean_locus_editing = mean(fig2d_mean_editing_by_locus$mean_fraction_edited)


fig2d_rbind$locus = as.factor(fig2d_rbind$locus)
fig2d_rbind[, locus_recolor := factor(
  locus,
  levels=unique(locus)[c(4,11,1,2,3,5,6,7,8,9,10,12)])]

color_2d <- c('gyrA' = '#F8766D', 'rpoB' = '#00B4F0',
              'sucB' = '#FF64B0',
               'mreC' = '#DE8C00', 'murF' = '#00BA38', 'lolA' = '#619CFF',
               'lpxK' = '#B79F00','fabH' = '#00C08B','hemA' = '#C77CFF',
               'fliN' = '#7CAE00','hda' = '#00BFC4','coaD' = '#F564E3')

####
mean_locus_editing = mean(fig2d_mean_editing_by_locus$mean_fraction_edited)

```


```{r h10_redo}
fig_1D <- ggplot(fig2d_rbind, aes(x= locus, y=fraction_edited, color=locus)) + 
  #geom_beeswarm(size = 4, priority = "random",
  #              cex = 4, dodge.width=2,
  #              alpha = 0.7, stroke = 0) +
  geom_quasirandom(method='pseudorandom', size = 3, alpha = 0.7, stroke = 0)+
  geom_hline(yintercept=mean_locus_editing, linetype=2, size=0.5) +
  theme_linedraw() +
  coord_cartesian(ylim=c(.5, .95)) +
  scale_y_continuous(breaks = c(.5, .6, .7, .8, .9),
                     labels = c("50%","60%", "70%", "80%", "90%")) +
  labs(
    y = "Fraction Edited") +
  facet_grid(~label, scales = 'free_x', space = 'free_x',
             switch = 'both') + #scales='free_x'
  scale_colour_manual(values = color_2d) +
  theme(
    axis.text.x = element_text(angle = 60, vjust = 1, hjust=1,
                               size = fig_font_size),
    axis.text.y = element_text(size = fig_font_size),
    axis.title.y = element_text(size = fig_title_font_size),
    axis.title.x = element_blank(),
    strip.background=element_blank(),
    panel.grid.minor=element_blank(),
    panel.grid.major.x=element_blank(),
    panel.border=element_blank(),
    axis.line=element_line(),
    legend.position = 'none')

fig_1D
```
```{r save_figure}
fig_width = 3.5
fig_height = 3.6 #edit of X,y axis makes it shorter

ggsave(here::here('fig_2D_quasi2.pdf'), plot=fig_1D, width=fig_width, 
       height=fig_height, units='in', dpi=300, device = cairo_pdf)
#ggplotly()
```


