---
title: "Fig_4c_possibly"
author: "Max Schubert"
date: "6/18/2019"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(data.table)
#library(stringr)
library(plyr)
library(gridExtra)
require(ggplot2)
require(data.table)
require(scales)
library(ggpubr)
library(plotly)
#dan's suggestion for plotting:
library(ggrepel)

parent_folder = here::here('fig_4c_possibly')

```

```{r get data}
#can update when i get the new stuff
test_table = "variants_fromT1000T1500.csv"

test_path = paste(parent_folder, test_table, sep = "/")
test_file_dt = data.table(fread(test_path))

#ditch columns with na's
#cutting down table to only a few columns
test_file_dt = Filter(function(x)!all(is.na(x)), test_file_dt) #kills columns with all NA
test_file_trimmed_dt = test_file_dt[,c( "UID", "POSITION",
                                        "REF", "ALT",
                                        "EXPERIMENT_SAMPLE_LABEL", "DP",
                                        "INFO_EFF_CONTEXT","INFO_EFF_EFFECT",
                                        "INFO_EFF_AA","INFO_EFF_GT",
                                        "INFO_EFF_CLASS", "INFO_EFF_GENE")] #keeps only specified columns

```

```{r plot_1000}
ggplot(test_file_trimmed_dt[EXPERIMENT_SAMPLE_LABEL == "T1000_cutadapt"], aes(x=POSITION, y=DP, color=EXPERIMENT_SAMPLE_LABEL, label = INFO_EFF_GENE)) + 
  geom_col(position = position_dodge(width = 0.9), size = 2, alpha = 0.5) +
  theme_bw() +
  geom_text(vjust = 0, data = subset(test_file_trimmed_dt, DP > 1000 & EXPERIMENT_SAMPLE_LABEL == "T1000_cutadapt")) +
  theme(text = element_text(size=18)) #+
  #coord_polar()
  #facet_wrap(~EXPERIMENT_SAMPLE_LABEL, nrow=2, ncol=1, scales = 'free_y') +
  #coord_polar() # +
  #geom_text_repel() +
  #labs(
  #  x = "Mutations, by genomic coordinate",
  #  y = "Depth of coverage",
  #  title = "Loci conferring TMP resistance in sheared gDNA RLR",
  #  color = "Sample")
```

#8/29/2019 trying new stuff
```{r plot_new}

#convert counts to relative measure of frequency
test_file_trimmed_dt[, snp_obs := sum(DP), by= 'EXPERIMENT_SAMPLE_LABEL']
test_file_trimmed_dt[, snp_freq := DP/snp_obs]

ggplot(test_file_trimmed_dt, aes(x=POSITION, y=snp_freq, label = INFO_EFF_GENE)) +
  geom_point(size = 2, alpha = 0.5) +
  #geom_col(position = position_dodge(), size = 2, alpha = 0.5) +
  theme_bw() +
  geom_text(vjust = 0, data = subset(test_file_trimmed_dt, snp_freq > .02 )) +
  theme(text = element_text(size=18)) + #+
  #coord_polar()
  facet_wrap(~EXPERIMENT_SAMPLE_LABEL, nrow=2, ncol=1, scales = 'free_y') #+
  #coord_polar() # +
  #geom_text_repel() +
  #labs(
  #  x = "Mutations, by genomic coordinate",
  #  y = "Depth of coverage",
  #  title = "Loci conferring TMP resistance in sheared gDNA RLR",
  #  color = "Sample")
```


```{r alternative_plot}
ggplot(test_file_trimmed_dt[EXPERIMENT_SAMPLE_LABEL == "T1500_cutadapt"], aes(x=POSITION, y=DP, color=INFO_EFF_EFFECT, label = INFO_EFF_GENE)) + 
  geom_col(position = position_dodge(width = 0.9), size = 2, alpha = 0.5) +
  theme_bw() +
  geom_text(vjust = 0, data = subset(test_file_trimmed_dt, DP > 1000 & EXPERIMENT_SAMPLE_LABEL == "T1500_cutadapt")) +
  theme(text = element_text(size=18)) #+
  #coord_polar()
  #facet_wrap(~EXPERIMENT_SAMPLE_LABEL, nrow=2, ncol=1, scales = 'free_y') +
  #coord_polar() # +
  #geom_text_repel() +
  #labs(
  #  x = "Mutations, by genomic coordinate",
  #  y = "Depth of coverage",
  #  title = "Loci conferring TMP resistance in sheared gDNA RLR",
  #  color = "Sample")
```

```{r zoom_plot}
#for zoom in on folA locus
zoomed_dt = test_file_trimmed_dt[POSITION < 60000]
zoomed_dt = subset(zoomed_dt, EXPERIMENT_SAMPLE_LABEL = "291_303")


ggplot(zoomed_dt, aes(x=POSITION, y=DP)) + 
  geom_col(position = position_dodge(width = 0.9), fill = "#FF6666", size =2 ) +
  theme_bw() +
  scale_y_continuous(trans = log10_trans()) +
  theme(text = element_text(size=24)) +
  theme_bw() +
  labs(
    x = "Mutations, shown at their position on the chromosome",
    y = "Depth of coverage",
    title = "Loci conferring TMP resistance in sheared gDNA RLR")

#optional, to inspect plot
#ggplotly()


#version that is less zoomed
zoomed_dt = test_file_trimmed_dt[POSITION < 1000000]
zoomed_dt = subset(zoomed_dt, EXPERIMENT_SAMPLE_LABEL = "291_303")


ggplot(zoomed_dt, aes(x=POSITION, y=DP)) + 
  geom_col(position = position_dodge(width = 0.9), fill = "#FF6666", size = 100) +
  theme_bw() +
  scale_y_continuous(trans = log10_trans()) +
  theme(text = element_text(size=24)) +
  theme_bw() +
  labs(
    x = "Mutations, shown at their position on the chromosome",
    y = "Depth of coverage",
    title = "Loci conferring TMP resistance in sheared gDNA RLR")

```


#############

# G10 data


```{r get_G10_data}
#can update when i get the new stuff
G10_table = "G10_new_variants.csv"
G10_path = paste(parent_folder, G10_table, sep = "/")
G10_dt = data.table(fread(G10_path))

#ditch columns with na's
#cutting down table to only a few columns
G10_dt = Filter(function(x)!all(is.na(x)), G10_dt) #kills columns with all NA
G10_dt_trimmed = G10_dt[,c( "UID", "POSITION",
                                        "REF", "ALT",
                                        "EXPERIMENT_SAMPLE_LABEL", "DP",
                                        "INFO_EFF_CONTEXT","INFO_EFF_EFFECT",
                                        "INFO_EFF_AA","INFO_EFF_GT",
                                        "INFO_EFF_CLASS", "INFO_EFF_GENE")] #keeps only specified columns

```

```{r get_muts}
#load 38a and 38B induced
induced_all = G10_dt[EXPERIMENT_SAMPLE_LABEL %in% c("38A_induced", "38B_induced", "40A_induced", "40B_induced")]

induced_all[, total_induced_depth := sum(DP), by = c("UID")]
induced_all = induced_all[,c( "UID", "POSITION",
                                        "REF", "ALT",
                                        "EXPERIMENT_SAMPLE_LABEL", "DP",
                                        "INFO_EFF_CONTEXT","INFO_EFF_EFFECT",
                                        "INFO_EFF_AA","INFO_EFF_GT",
                                        "INFO_EFF_CLASS", "INFO_EFF_GENE", "total_induced_depth")]

induced_snps = induced_all[EXPERIMENT_SAMPLE_LABEL %in% c("38A_induced")]
induced_snps= induced_snps[,c( "UID", "POSITION",
                                        "REF", "ALT",
                                        "INFO_EFF_CONTEXT","INFO_EFF_EFFECT",
                                        "INFO_EFF_AA","INFO_EFF_GT",
                                        "INFO_EFF_CLASS", "INFO_EFF_GENE", "total_induced_depth")]
```


```{r plot_40}
dt40plus = G10_dt_trimmed[EXPERIMENT_SAMPLE_LABEL %in% c("40A", "40B")]

#convert counts to relative measure of frequency
dt40plus[, snp_obs := sum(DP), by= 'EXPERIMENT_SAMPLE_LABEL']
dt40plus[, snp_freq := DP/snp_obs]



plot_40 <- ggplot(dt40plus, aes(x=POSITION, y=snp_freq, color=EXPERIMENT_SAMPLE_LABEL, label = INFO_EFF_GENE)) + 
  geom_col(position = position_dodge(width = 0.9), size = 2, alpha = 0.5) +
  theme_bw() +
  #scale_y_continuous(trans = log10_trans() ) + #limit=c(0.0005, 8), 
                     #breaks = c(.001, .01, .1, .3, 1, 3, 10, 100, 1000)) +
  geom_text(vjust = 0, data = subset(dt40plus_filter, snp_freq >= .01)) +
  theme(text = element_text(size=18),
        legend.position = 'none') +
  #coord_polar()
  facet_wrap(~EXPERIMENT_SAMPLE_LABEL, nrow=1, ncol=2)
  #coord_polar() # +
  #geom_text_repel() +
  #labs(
  #  x = "Mutations, by genomic coordinate",
  #  y = "Depth of coverage",
  #  title = "Loci conferring TMP resistance in sheared gDNA RLR",
  #  color = "Sample")

plot_40
```

#38 are the samples with beta recombinase
```{r plot_38}
dt38plus = G10_dt_trimmed[EXPERIMENT_SAMPLE_LABEL %in% c("38Aplus", "38Bplus")]

#convert counts to relative measure of frequency
dt38plus[, snp_obs := sum(DP), by= 'EXPERIMENT_SAMPLE_LABEL']
dt38plus[, snp_freq := DP/snp_obs]

#remove folA snp and see what's left:
#dt40plus_filter = dt40plus[UID != "aa76968f"]
#dt40plus_filter = dt40plus_filter[snp_freq > .001]

#dt40plus_faves = dt40plus_filter[INFO_EFF_GENE %in% c('cstA', 'igaA', 'uvrD')]

plot_38 <- ggplot(dt38plus, aes(x=POSITION, y=snp_freq, color=EXPERIMENT_SAMPLE_LABEL, label = INFO_EFF_GENE)) + 
  geom_col(position = position_dodge(width = 0.9), size = 2, alpha = 0.5) +
  theme_bw() +
  #scale_y_continuous(trans = log10_trans() ) + #limit=c(0.0005, 8), 
                     #breaks = c(.001, .01, .1, .3, 1, 3, 10, 100, 1000)) +
  geom_text(vjust = 0, data = subset(dt38plus, snp_freq >= .01)) +
  theme(text = element_text(size=18),
        legend.position = 'none') +
  #coord_polar()
  facet_wrap(~EXPERIMENT_SAMPLE_LABEL, nrow=1, ncol=2)
  #coord_polar() # +
  #geom_text_repel() +
  #labs(
  #  x = "Mutations, by genomic coordinate",
  #  y = "Depth of coverage",
  #  title = "Loci conferring TMP resistance in sheared gDNA RLR",
  #  color = "Sample")
plot_38
```

```{r}
#plot both in grid
grid.arrange(
  grobs = list(plot_38, plot_40),
  #widths = c(2, 1, 1),
  layout_matrix = rbind(c(1, 1),
                        c(2, 2))
)
```

### what's in there besides fola?

```{r no_folA}
#remove folA snp and see what's left:
dt40plus_filter = dt40plus[UID != "aa76968f"]
#dt40plus_filter = dt40plus_filter[snp_freq > .001]

plot_40_filter <- ggplot(dt40plus_filter, aes(x=POSITION, y=snp_freq, color=EXPERIMENT_SAMPLE_LABEL, label = INFO_EFF_GENE)) + 
  geom_col(position = position_dodge(width = 0.9), size = 2, alpha = 0.5) +
  theme_bw() +

  geom_text(vjust = 0, data = subset(dt40plus_filter, snp_freq >= .01)) +
  theme(text = element_text(size=18),
        legend.position = 'none') +
  facet_wrap(~EXPERIMENT_SAMPLE_LABEL, nrow=1, ncol=2)


dt38plus_filter = dt38plus[UID != "aa76968f"]
plot_38_filter <- ggplot(dt38plus_filter, aes(x=POSITION, y=snp_freq, color=EXPERIMENT_SAMPLE_LABEL, label = INFO_EFF_GENE)) + 
  geom_col(position = position_dodge(width = 0.9), size = 2, alpha = 0.5) +
  theme_bw() +
  scale_y_continuous(limit=c(0, .25) )+
                     #breaks = c(.001, .01, .1, .3, 1, 3, 10, 100, 1000)) +
  geom_text(vjust = 0, data = subset(dt38plus, snp_freq >= .01)) +
  theme(text = element_text(size=18),
        legend.position = 'none') +
  #coord_polar()
  facet_wrap(~EXPERIMENT_SAMPLE_LABEL, nrow=1, ncol=2)


#plot both in grid
grid.arrange(
  grobs = list(plot_38_filter, plot_40_filter),
  #widths = c(2, 1, 1),
  layout_matrix = rbind(c(1, 1),
                        c(2, 2))
)


#closer look at 40B
ggplot(dt40plus_filter[EXPERIMENT_SAMPLE_LABEL == "40B"], aes(x=POSITION, y=snp_freq, color=INFO_EFF_EFFECT, label = INFO_EFF_GENE)) + 
  geom_col(position = position_dodge(width = 0.9), size = 2, alpha = 0.5) +
  theme_bw() +

  geom_text(vjust = 0, data = subset(dt40plus_filter, snp_freq >= .002)) +
  theme(text = element_text(size=18))
        
        
        legend.position = 'none') #+
  facet_wrap(~EXPERIMENT_SAMPLE_LABEL, nrow=1, ncol=2)


```


