---
title: "mut_counts"
author: "Max Schubert"
date: "2/2/2021"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}

knitr::opts_chunk$set(echo = TRUE)

library(data.table)
library(stringr)
library(plyr)
library(gridExtra)
require(ggplot2)
require(data.table)
require(scales)
library(ggpubr)
library(plotly)
#library(dplyr)

library(ggbeeswarm) #first install.packages("ggbeeswarm")

# mkdir fig_muts
parent_folder <- here::here('fig_nu_muts')
```


```{r get_variants}
snp_data_name = "zz4_variants_GT2.csv" #variants with GT = 2 flag, high confidence
snp_data_file = paste(parent_folder, snp_data_name, sep = "/")
snp_data_dt <- data.table(fread(snp_data_file))

#remove SNPs flagged for known issues, like low coverage, low mapping quality, or nonunique alignments
snp_data_dt= snp_data_dt[VARIANT_SET_LABEL == '[None]']
snp_data_dt= snp_data_dt[DP > 5]
snp_data_dt= snp_data_dt[INFO_AF > 0.85]

#remove snps in the araBAD,C or bioB regions, which are deleted and/or plasmid-borne, causing artifacts
snp_data_dt= snp_data_dt[POSITION < 67500 | POSITION > 71300]
snp_data_dt= snp_data_dt[POSITION < 807000 | POSITION > 887500]




#trim extraneous columns
snp_data_dt[, c("VARIANT_SET_LABEL", "INFO_PQR", "INFO_QR", "INFO_NS", "INFO_ODDS",  "UID", "INFO_EPPR", "INFO_LOF", "INFO_DPB", "INFO_GTI", "IS_SV",
                   "INFO_MQMR", "INFO_RPPR", "INFO_NMD","INFO_PRO", "INFO_RO", "INFO_SRF",
                   "INFO_SRP", "INFO_SRR", "RO", "GT", "GQ",  "AO", "QA", "GL",  "QR", "INFO_MQM",
                   "INFO_ABP", "INFO_TECHNOLOGY.ILLUMINA", "INFO_EPP", "INFO_PQA", "INFO_RUN",
                   "INFO_DPRA", "INFO_QA", "INFO_EFF_EFFECT", "INFO_TYPE", "INFO_PAO", "INFO_CIGAR",
                   "INFO_PAIRED", "INFO_AO", "INFO_SAP", "INFO_SAR", "INFO_RPR", "INFO_RPP", "INFO_AC",
                   "INFO_AB",  "INFO_EFF_TRLEN", "INFO_SAF", "INFO_EFF_AA", "INFO_RPL",  "INFO_EFF_WARN",
                   "INFO_EFF_ERR", "INFO_EFF_GT", "INFO_EFF_CLASS", "INFO_MEANALT", "INFO_EFF_RANK",
                   "INFO_EFF_CODING", "INFO_EFF_TR", "PHRED_ENCODING", "CHROMOSOME", "INFO_DP",
                   "INFO_NUMALT", "INFO_AN", "INFO_LEN", "IS_HET", "INFO_EFF_BIOTYPE") := NULL]


```

```{r filter_parent_variants}
snp446 = snp_data_dt[EXPERIMENT_SAMPLE_LABEL == "446"]
snp453 = snp_data_dt[EXPERIMENT_SAMPLE_LABEL == "453"]

mutlist_446 = snp446$POSITION
mutlist_453 = snp453$POSITION


####### DBG incomplete- remove all muts with position found in 453 or 446

#snp446new = snp446[POSITION %IN% mutlist_446]

```


```{r further_filter_variants}
#remove SNPs altered intentionally in ancestral strain, or present in retron donors
snp_data_dt= snp_data_dt[POSITION != 2082767] #intentional sbcB
snp_data_dt= snp_data_dt[POSITION != 2082772] #intentional sbcB
snp_data_dt= snp_data_dt[POSITION != 3038080] #intentional recJ

#remove SNPs aberrantly called 

```


```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
