---
title: "fig_4_sushi"
author: "Max Schubert"
date: "10/17/2019"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(data.table)
#library(stringr)
library(plyr)
library(gridExtra)
require(ggplot2)
require(data.table)
require(scales)
library(ggpubr)
library(plotly)
#dan's suggestion for plotting:
library(ggrepel)

#for plotting
library(cowplot)
library(gridGraphics) #first install.packages('gridGraphics')

parent_folder = here::here('fig_4_sushi') #TODO UPDATE WITH REAL NAME
source(here::here('constants.R'))
```

```{r import_data}
#TODO: upload to git
bedpath = "/Users/Max_Schubert/dropbox_hms/active_seq_reads/jan_2019_nextseq/FC_04592/Unaligned_1234_PF_mm1/Data/merged_replicates/"


#import depth
AB3_folA_nonuniq_dp = read.delim(paste(bedpath, "AB3_folA_nonuniq_dp", sep = ""), sep = "\t")
AB4_folA_nonuniq_dp = read.delim(paste(bedpath, "AB4_folA_nonuniq_dp", sep = ""), sep = "\t")
WT_folA_dp = read.delim(paste(bedpath, "s2_folA_dp", sep = ""), sep = "\t")
names(AB3_folA_nonuniq_dp) <- c('chr','pos','dp')
names(AB4_folA_nonuniq_dp) <- c('chr','pos','dp')
names(WT_folA_dp) <- c('chr','pos','dp')
#label and bind two replicates
AB3_folA_nonuniq_dp$replicate = "A"
AB4_folA_nonuniq_dp$replicate = "B"
WT_folA_dp$replicate = "WT"
dp_folA_region = rbind(AB3_folA_nonuniq_dp, AB4_folA_nonuniq_dp, WT_folA_dp)
rm(AB3_folA_nonuniq_dp, AB4_folA_nonuniq_dp, WT_folA_dp) #remove original files

#trim outer 50bp off
dp_folA_region = data.table(dp_folA_region)
dp_folA_region <- dp_folA_region[pos > 48050]
dp_folA_region <- dp_folA_region[pos < 50950]

#import stranded counted bed
AB3_cutadapt_aln_folAzoom.bedgraph = read.delim(paste(bedpath, "AB3_cutadapt_aln_folAzoom.bedgraph", sep = ""), sep = "")
names(AB3_cutadapt_aln_folAzoom.bedgraph) <- c('chr','start', 'end','strand','count')
AB3_cutadapt_aln_folAzoom.bedgraph = data.table(AB3_cutadapt_aln_folAzoom.bedgraph)
dt_folA_bedgraph = data.table(AB3_cutadapt_aln_folAzoom.bedgraph)

```

```{r plot_depth_ggplot}
#change order of replicates, set colors
dp_folA_region$replicate = factor(dp_folA_region$replicate, levels = c('WT','B','A'))
colormapping = c(A='#F8766D', B='#00BFC4', WT='#DCDCDC')

##### LOG VERSION
depth_plot <- ggplot(dp_folA_region) + 
  theme_bw() +
  labs( y = "Coverage Depth",
        x = "Genome Position") +
  scale_x_continuous(limits = c(48000, 51000)) +
  scale_y_continuous(trans = log_trans(base=10),
                     breaks = c(0, 10, 100, 1000, 10000,100000)) +
  theme(
    strip.background=element_blank(), #strip.placement='outside',
    strip.text=element_text(color='black'), 
    panel.border=element_blank(),
    axis.line=element_line(),
    legend.position = c(0.1, .8)) +
  geom_density((aes(y=dp, x=pos, color = replicate,
                   fill = replicate, alpha = 0.1)),
                stat = "identity") +
  scale_fill_manual(values=colormapping) +
  scale_color_manual(values=colormapping) +
  scale_alpha(guide = FALSE) +
  #add vert lines for snps
  geom_segment(x = 49765, xend = 49765, y = 0, yend = log10(max(dp_folA_region[pos == 49765]$dp)),
             color = "black", size = 0.2) +
  geom_segment(x = 49903, xend = 49903, y = 0, yend = log10(max(dp_folA_region[pos == 49903]$dp)),
             color = "black", size = 0.2) +
  geom_segment(x = 50280, xend = 50280, y = 0, yend = log10(max(dp_folA_region[pos == 50280]$dp)),
             color = "black", size = 0.2) +
  #add snp annotations
  annotate("text", x= 49765 - 250, y = max(dp_folA_region[pos == 49765]$dp), label = "49765: C > T") +
  annotate("text", x= 49903 + 250, y = max(dp_folA_region[pos == 49903]$dp), label = "49903: T > G") +
  annotate("text", x= 50280 + 250, y = max(dp_folA_region[pos == 50280]$dp - 10), label = "50280: T > G")


#50280	U00096.2	T	G
depth_plot

```

#plot unique donors contibuting to signal
###filtering to those observed more than 1000 times
```{r col_plot}
#gen uniq IDs for each row, trim data
data = data.table(dt_folA_bedgraph)
data[, serial := .I]
datatrim <- data[count > 1000]
datatrim$plotrow <- c(1,2,3,4,5,6,7,1,2,3,4,5)

#plot segments
#this output gets cleaned up in illustrator
col_plot <- ggplot(datatrim) + 
  theme_bw() +
  labs(x = "Genome Position") +
  scale_x_continuous(limits = c(48000, 51000)) +
  scale_y_continuous(breaks = c(1,2,3),
                     labels = c("1.000", "2.000", "3.000")) +
  theme(
    strip.background=element_blank(), #strip.placement='outside',
    strip.text=element_text(color='black'), 
    panel.border=element_blank(),
    axis.line=element_line(),
    legend.position = c(0.1, .5)) +
  #  legend.position = c(0.1, .8)) +
  geom_segment((aes(y=plotrow/2, yend = plotrow/2, size = 2,
                    x=start, xend = end, group = serial,
                   color = count)),
                stat = "identity") +
  scale_color_gradient(low = "blue", high = "red") +
  scale_size(guide = FALSE)
col_plot

```
```{r facet_both_together}
facetted <- plot_grid(depth_plot, col_plot, nrow = 2)

ggsave('fig_4C_sushi.pdf', plot=facetted, width=4, height=5, units='in', dpi=300, useDingbats = FALSE)
```




``` {r bedplot_cruft}
legend("topright",inset=0,legend=c("reverse","forward"),fill=SushiColors(2)(2),border=SushiColors(2)(2),text.font=2,cex=0.75)




> labelgenome(chrom,chromstart,chromend,n=2,scale="Kb")> legend("topright",inset=0,legend=c("reverse","forward"),fill=SushiColors(2)(2),border=SushiColors(2)(2),text.font=2,cex=0.75)



#erase strandedness of bedgraph
tester = dt_folA_bedgraph[start > end]
dt_folA_bedgraph_directionless <- dt_folA_bedgraph[start > end, end := start & start := end]


#plot both replicates
plotBedgraph(dt_folA_bedgraph,chrom,chromstart,chromend,transparency=.50,color=SushiColors(2)(2)[1])
labelgenome(chrom,chromstart,chromend,n=4,scale="Kb")
mtext("Read Depth",side=2,line=1.75,cex=1,font=2)
axis(side=1,las=2,tcl=.2)


plotBedgraph(dt_folA_bed,chrom,chromstart,chromend,transparency=.50,color=SushiColors(2)(2)[1])

> plotBedgraph(Sushi_DNaseI.bedgraph,chrom,chromstart,chromend,transparency=.50,color=SushiColors(2)(2)[2],overlay=TRUE,rescaleoverlay=TRUE)> labelgenome(chrom,chromstart,chromend,n=3,scale="Kb")

dt[count > 500] #filter some junk out
fwd <- dt[end > start, strand := "+"]
rev <- dt[start > end, strand := "-"]
#dt2[,magnitude := as.integer(count/10000)]
#dt2[,rel_magnitude := count/max(dt2$count)]


#first plot
plotBedgraph(dt_folA_bed,"U00096.2",49000,51000,
             colorbycol= SushiColors(5))
labelgenome("U00096.2",49000,51000,n=2,scale="Kb")
p1 <- recordPlot()

#second plot
plotBed(beddata = dt2,
        chrom = "U00096.2",
        chromstart = 49000,chromend = 51000,
        colorby = dt2$magnitude,
        colorbycol = SushiColors(5),
        row = "auto")
labelgenome("U00096.2",49000,51000,n=2,scale="Kb")
legend("topright",inset=0,legend=c("<10k","20k","30k","40k", "50k"),
       fill=SushiColors(5)(5),border=SushiColors(5)(5),
       text.font=2,cex=0.75)
p2 <- recordPlot()

plot_grid(p1, p2, nrow = 2)
```

```{r cruft_tests}
#TODO delete

## ggplot segment version
data = dt2
data






#test:
#Sushi_data = data(package ='Sushi')
#data(list = Sushi_data$results[,3])
#try bedgraph
#plotBedgraph(AB3_bed,"U00096.2",45000,55000,colorbycol= SushiColors(5))

#try plotbed
dt = data.table(AB3_cutadapt_aln_folAzoom.bedgraph)
#build multi-panel plot

#par(mfrow=c(2,1)) #figure margins too large ?
#par(fig=c(0,1,0,0.5), new=TRUE) #invalid graphics state.... what?
#par(fig=c(0,1,0.5, 1), new=TRUE)

layout(matrix(c(1,2), 1, 1, byrow = TRUE),
   widths=c(3), heights=c(1))
##### LINEAR VERSION
depth_plot <- ggplot(dp_folA_region) + 
  theme_bw() +
  labs( y = "Coverage Depth",
        x = "Genome Position") +
  #scale_y_continuous(trans = log_trans(base=10),
  #                   breaks = c(0, 10, 100, 1000, 10000,100000)) +
  theme(
    strip.background=element_blank(), #strip.placement='outside',
    strip.text=element_text(color='black'), 
    panel.border=element_blank(),
    axis.line=element_line(),
    legend.position = c(0.1, .8)) +
  geom_density(aes(y=dp, x=pos, color = replicate,
                   fill = replicate, alpha = 0.01),
                stat = "identity") +
  scale_alpha(guide = FALSE) +
  #add vert lines for snps
  geom_segment(x = 49765, xend = 49765, y = 0, yend = max(dp_folA_region[pos == 49765]$dp),
             color = "red", size = 0.2) +
  geom_segment(x = 49903, xend = 49903, y = 0, yend = max(dp_folA_region[pos == 49903]$dp),
             color = "red", size = 0.2) #+
  #geom_segment(x = 50280, xend = 50280, y = 0, yend = log10(max(dp_folA_region[pos == 50280]$dp)),
  #           color = "red", size = 0.2)

depth_plot


########### looking at dropped region
depth_plot <- ggplot(dp_folA_region[pos %between% c(49300, 49400)]) + 
  theme_bw() +
  labs( y = "Coverage Depth",
        x = "Genome Position") +
  #scale_y_continuous(trans = log_trans(base=10),
  #                   breaks = c(0, 10, 100, 1000, 10000,100000)) +
  theme(
    strip.background=element_blank(), #strip.placement='outside',
    strip.text=element_text(color='black'), 
    panel.border=element_blank(),
    axis.line=element_line(),
    legend.position = c(0.1, .8)) +
  geom_density(aes(y=dp, x=pos, color = replicate,
                   fill = replicate, alpha = 0.01),
                stat = "identity") +
  scale_alpha(guide = FALSE)


depth_plot
```

######## ggplot version of bedplot

```{r prep_ggplot_data}
#gen uniq IDs for each row, trim data
data = dt_folA_bedgraph
data[, serial := .I]
data[count > 100]

#rand_y = quote(runif(1,min = 1, max = 100))

#determine rows for plotting in ggplot:
#function:
checkrow <- function(data,alldata,maxrows,wiggle=0)
  {
    for (row in (1:maxrows))
    {
      thestart = min(as.numeric(data[c(2,3)])) - wiggle
      thestop  = max(as.numeric(data[c(2,3)])) + wiggle
      
      if (nrow(alldata[which(alldata$plotrow == row &
                               ((thestart >= alldata[,2] & thestart <= alldata[,3]) |
                                  (thestop >= alldata[,2] & thestop <= alldata[,3]) |
                                  (thestart <= alldata[,2] & thestop >= alldata[,3]))),]) == 0)
      {
        return (row)
      }
    }
  }

## alt version:
checkrow_alt <- function(data,alldata,maxrows,wiggle=0)
  {
    for (row in (1:maxrows))
    {
      thestart = min(as.numeric(data$start)) - wiggle
      thestop  = max(as.numeric(data$end)) + wiggle
      
      if (nrow(alldata[which(alldata$plotrow == row &
                               ((thestart >= alldata$start & thestart <= alldata$end) |
                                  (thestop >= alldata$start & thestop <= alldata$end) |
                                  (thestart <= alldata$start & thestop >= alldata$end))),]) == 0)
      {
        return (row)
      }
    }
  }
  

#usage:
maxrows = 1000
beddata = dt2
wiggle = 0

for (i in (1:nrow(beddata)))
      {
        beddata[i,]$plotrow = checkrow_alt(beddata[i,],beddata,maxrows,wiggle=wiggle)
      }


```


```{r ggplot_bedgraph}

#plot segments
col_plot <- ggplot(data) + 
  theme_bw() +
  labs(x = "Genome Position") +
  #scale_y_continuous(trans = log_trans(base=10),
  #                   breaks = c(0, 10, 100, 1000, 10000,100000)) +
  theme(
    strip.background=element_blank(), #strip.placement='outside',
    strip.text=element_text(color='black'), 
    panel.border=element_blank(),
    axis.line=element_line()) +
  #  legend.position = c(0.1, .8)) +
  geom_segment((aes(y=serial, yend = serial + 1,#generating as random for now...
                    x=start, xend = end, group = serial,
                   color = count, alpha = 1)),
                stat = "identity") 
col_plot

+
  scale_fill_manual(values=colormapping) +
  scale_color_manual(values=colormapping) +
  scale_alpha(guide = FALSE) +
  #add vert lines for snps
  geom_segment(x = 49765, xend = 49765, y = 0, yend = log10(max(dp_folA_region[pos == 49765]$dp)),
             color = "black", size = 0.2) +
  geom_segment(x = 49903, xend = 49903, y = 0, yend = log10(max(dp_folA_region[pos == 49903]$dp)),
             color = "black", size = 0.2) +
  #add snp annotation
  annotate("text", x= 49765 - 250, y = max(dp_folA_region[pos == 49765]$dp), label = "49765: C > T") +
  annotate("text", x= 49903 + 250, y = max(dp_folA_region[pos == 49903]$dp), label = "49903: T > G")

depth_plot


#second plot

```

#test:
#library(rtracklayer) #BiocManager::install("rtracklayer")
#library('Sushi') #BiocManager::install("Sushi")
#library(ggridges) #install.packages('ggridges')



```{r plot_folA_bed}
#constants
chrom = "U00096.2"
chromstart = 49000
chromend = 51000
count_thresh = 1000

#pre-process table
dt2 <- AB3_cutadapt_aln_folAzoom.bedgraph[count > count_thresh]
dt2[, strand := as.character(strand)]
dt2[strand == "+", strand := "1"]
dt2[strand == "-", strand := "-1"]
dt2[,strand := as.integer(strand)]
dt2$new_row = dt2$strand #cheat to get strand into col 6
dt2[,magnitude := as.integer(count/10000)]
colorpalette = SushiColors(5)

#plotting
plotBed(beddata = dt2,
        chrom = chrom,
        chromstart = chromstart,
        chromend = chromend,
        colorby    = dt2$strand,
        #col = maptocolors(dt2$magnitude,colorpalette),
        colorbycol = SushiColors(2), #SushiColors(5)
        row  = "auto",wiggle=0.001, splitstrand=TRUE)
labelgenome(chrom,chromstart,chromend,n=10,scale="Kb")
bed_above_1000 <- recordPlot()

```
```{r facet_and_save}
facetted <- plot_grid(depth_plot, bed_above_1000, nrow = 2)

ggsave('fig_4C_sushi.pdf', plot=facetted, width=4, height=5, units='in', dpi=300, useDingbats = FALSE)
```

#bedname = "AB3_cutadapt_aln.bed"
#bedfile = paste(bedpath, bedname, sep = "")

#AB3_bed = read.delim(bedfile, sep = "\t")

#bed file of all reads
#AB3_cutadapt_aln_folAzoom.bed = read.delim(paste(bedpath, "AB3_cutadapt_aln_folAzoom.bed", sep = ""), sep = "\t")
#names(AB3_cutadapt_aln_folAzoom.bed) <- c('chr','start','end', 'strand')

#bedgraph of uniq-ed sequences
#AB3_cutadapt_aln_folAzoom.bedgraph = read.delim(paste(bedpath, "AB3_cutadapt_aln_folAzoom.bedgraph", sep = ""), sep = " ")
#names(AB3_cutadapt_aln_folAzoom.bedgraph) <- c('chr','start','end', 'count')

#mk dts
#dt_folA_bed = data.table(AB3_cutadapt_aln_folAzoom.bed)
