---
title: "fig_5C"
author: "Max Schubert"
date: "10/17/2019"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(data.table)
#library(stringr)
library(plyr)
library(gridExtra)
require(ggplot2)
require(data.table)
require(scales)
library(ggpubr)
#dan's suggestion for plotting:
library(ggrepel)
library(ggnewscale)

#for plotting
library(cowplot)
library(gridGraphics) #first install.packages('gridGraphics')

parent_folder = here::here('fig_5C/')
source(here::here('constants.R'))
```

```{r import_data}
bedpath = parent_folder

#import depth
AB3_folA_nonuniq_dp = read.delim(paste(bedpath, "AB3_folA_nonuniq_dp", sep = ""), sep = "\t")
AB4_folA_nonuniq_dp = read.delim(paste(bedpath, "AB4_folA_nonuniq_dp", sep = ""), sep = "\t")
WT_folA_dp = read.delim(paste(bedpath, "s2_folA_dp", sep = ""), sep = "\t")
names(AB3_folA_nonuniq_dp) <- c('chr','pos','dp')
names(AB4_folA_nonuniq_dp) <- c('chr','pos','dp')
names(WT_folA_dp) <- c('chr','pos','dp')
#label and bind two replicates
AB3_folA_nonuniq_dp$replicate = "A"
AB4_folA_nonuniq_dp$replicate = "B"
WT_folA_dp$replicate = "WT"
dp_folA_region = rbind(AB3_folA_nonuniq_dp, AB4_folA_nonuniq_dp, WT_folA_dp)
rm(AB3_folA_nonuniq_dp, AB4_folA_nonuniq_dp, WT_folA_dp) #remove original files

#trim outer 50bp off
dp_folA_region = data.table(dp_folA_region)
dp_folA_region <- dp_folA_region[pos > 48050]
dp_folA_region <- dp_folA_region[pos < 50950]

#import stranded counted bed
AB3_cutadapt_aln_folAzoom.bedgraph = read.delim(paste(bedpath, "AB3_cutadapt_aln_folAzoom.bedgraph", sep = ""), sep = "")
names(AB3_cutadapt_aln_folAzoom.bedgraph) <- c('chr','start', 'end','strand','count')
AB3_cutadapt_aln_folAzoom.bedgraph = data.table(AB3_cutadapt_aln_folAzoom.bedgraph)
dt_folA_bedgraph = data.table(AB3_cutadapt_aln_folAzoom.bedgraph)

```


#old version
```{r plot_depth_ggplot}
#change order of replicates, set colors
dp_folA_region$replicate = factor(dp_folA_region$replicate, levels = c('WT','B','A'))
colormapping = c(A='#F8766D', B='#00BFC4', WT='#DCDCDC')

##### LOG VERSION
depth_plot <- ggplot(dp_folA_region) + 
  theme_bw() +
  labs( y = "Coverage Depth",
        x = "Genome Position") +
  scale_x_continuous(limits = c(48000, 51000)) +
  scale_y_continuous(trans = log_trans(base=10),
                     breaks = c(0, 10, 100, 1000, 10000,100000)) +
  theme(
    strip.background=element_blank(), #strip.placement='outside',
    strip.text=element_text(color='black'), 
    panel.border=element_blank(),
    axis.line=element_line(),
    legend.position = c(0.1, .8)) +
  geom_density((aes(y=dp, x=pos, color = replicate,
                   fill = replicate, alpha = 0.1)),
                stat = "identity") +
  scale_fill_manual(values=colormapping) +
  scale_color_manual(values=colormapping) +
  scale_alpha(guide = FALSE) +
  #add vert lines for snps
  geom_segment(x = 49765, xend = 49765, y = 0, yend = log10(max(dp_folA_region[pos == 49765]$dp)),
             color = "black", size = 0.2) +
  geom_segment(x = 49903, xend = 49903, y = 0, yend = log10(max(dp_folA_region[pos == 49903]$dp)),
             color = "black", size = 0.2) +
  geom_segment(x = 50280, xend = 50280, y = 0, yend = log10(max(dp_folA_region[pos == 50280]$dp)),
             color = "black", size = 0.2) +
  #add snp annotations
  annotate("text", x= 49765 - 250, y = max(dp_folA_region[pos == 49765]$dp), label = "49765: C > T") +
  annotate("text", x= 49903 + 250, y = max(dp_folA_region[pos == 49903]$dp), label = "49903: T > G") +
  annotate("text", x= 50280 + 250, y = max(dp_folA_region[pos == 50280]$dp - 10), label = "50280: T > G")


#50280	U00096.2	T	G
depth_plot
```

### DBG, normalized vers
```{r set up figure}
replicate_cols <- brewer.pal(n=9, name='YlGnBu')[c(5,8)]
gene_cols <- brewer.pal(n=9, name='Greys')[c(5,4)]
donor_cols <- brewer.pal(n=9, name='YlGn')[c(3,9)]

#change order of replicates, set colors
dp_folA_region$replicate = factor(dp_folA_region$replicate, levels = c('WT','B','A'))
colormapping = c(A=replicate_cols[1], B=replicate_cols[2], WT='#888888')
dp_folA_region[, dp_norm := dp / max(dp), by='replicate']

snp_lines = data.table(
 pos= c(49765, 49903, 50280), 
 lbl = c('49765: C > T','49903: T > G','50280: T > G'))
snp_lines[, dp_norm := dp_folA_region[pos == .BY[1], max(dp_norm)], by='pos']

genome_start = 48050
genome_end = 50950
genome_breaks = c(48500, 49000, 49500, 50000, 50500)
genome_br_lbl = paste0(genome_breaks/1e3,'kb')

# polygon shapes
gene_y_pos = 1*10^1.3
gene_pos <- data.table(
  name=c('folA','kefC'), start=c(49823,48050), end=c(50302, 49631))


#gen uniq IDs for each row, trim data
data = data.table(dt_folA_bedgraph)
data[, serial := .I]
datatrim <- data[count > 1000]
datatrim$plotrow <- c(1,2,3,4,5,6,7,1,2,3,4,5)
```

```{r depth plot}
##### LOG VERSION
depth_plot <- ggplot(dp_folA_region) + 
  theme_bw() +
  geom_segment(data=gene_pos, 
    aes(x=start, xend=end, color=name,
        y=gene_y_pos, yend=gene_y_pos),
    size = 5, 
    arrow=arrow(length = unit(0.02, "npc"), type='closed'), 
    linejoin = 'mitre', lineend='butt') +  
  geom_text(data=gene_pos, aes(x=start, y=gene_y_pos, 
    label=name), hjust=-0.3, vjust=-0.8, size=fig_font_size/2) +
  scale_color_manual(values=gene_cols, guide=F) +
  new_scale_color() +
  labs( y = "Coverage Depth (/maximum)",
       x = "Genome Position") +
  scale_x_continuous(limits = c(genome_start, genome_end), expand=c(0,0), 
                     breaks=genome_breaks, labels=genome_br_lbl) +
  scale_y_continuous(trans = log_trans(base=10), 
                    breaks = c(.00001, 0.0001, 0.001, 0.01, 0.1, 1)) +
  coord_cartesian(xlim=c(genome_start, genome_end), ylim = c(.5e-5,gene_y_pos*1.5), clip='on') +
  theme(
    strip.background=element_blank(), #strip.placement='outside',
    strip.text=element_text(color='black'), 
    panel.border=element_blank(),
    axis.line=element_line(),
    axis.title.x=element_blank(),
    axis.ticks.x=element_blank(),
    axis.text.x = element_text(size = fig_font_size),
    axis.text.y = element_text(size = fig_font_size),
    axis.title.y = element_text(size = fig_title_font_size),
    panel.grid.minor=element_blank(),
    legend.position = c(0.08, .50)) +
  geom_line(aes(y=dp_norm, x=pos, color = replicate), size=1, alpha=0.8) +
  scale_fill_manual(values=colormapping) +
  scale_color_manual(values=colormapping) +
  scale_alpha(guide = FALSE) +
  #add vert lines for snps
  geom_segment(data=snp_lines, aes(x=pos, xend=pos, y=1e-6, yend=dp_norm), linetype=2) +
  #add snp annotations
  annotate("text", x= 49765, y = 10^(0.1), label = "49765: C>T", hjust=0, angle=45, size=fig_font_size/2) +
  annotate("text", x= 49903, y = 10^(-0.15), label = "49903: T>G", hjust=0, angle=45, size=fig_font_size/2) +
  annotate("text", x= 50280, y = 10^(-0.3), label = "50280: T>G", hjust=0, angle=45, size=fig_font_size/2)


#50280	U00096.2	T	G
depth_plot
```
```{r segment plot}
#plot segments
#this output gets cleaned up in illustrator
col_plot <- ggplot(datatrim) + 
  theme_bw() +
  labs(x = "Genome Position") +
  scale_y_continuous(name='Unique Retrons') + 
  scale_x_continuous(limits = c(genome_start, genome_end), expand=c(0,0), 
                     breaks=genome_breaks, labels=genome_br_lbl) +
  coord_cartesian(xlim=c(genome_start, genome_end), clip='off') +
  theme(
    strip.background=element_blank(), #strip.placement='outside',
    strip.text=element_text(color='black'), #, size = fig_font_size),
    axis.line.y=element_blank(),
    axis.text=element_blank(),
    axis.title.x=element_blank(),
    axis.title.y=element_text(size = fig_title_font_size),
    axis.ticks=element_blank(),
    panel.grid.major.y=element_blank(),
    panel.grid.minor=element_blank(),
    panel.border=element_blank(),
    legend.position = c(0.9, .55),
    legend.title = element_text(size = fig_font_size)) +
  #  legend.position = c(0.1, .8)) +
  geom_segment(aes(y=plotrow/2, yend = plotrow/2,
                    x=start, xend = end, group = serial,
                   color = count),
                size = 4,
                stat = "identity") +
  scale_color_gradientn(
    name='# Reads', colors=donor_cols, limits=c(1000,60000), 
    breaks=c(1,2,3,4,5)*10^4, labels=paste0(c(1,2,3,4,5),' x 10^5'),
    guide=guide_legend(keyheight=0.9)) +
  scale_size(guide = FALSE) +
  geom_vline(data=snp_lines[1:2], aes(xintercept=pos), linetype=2)
col_plot
```
```{r combined}
combined_5c <- plot_grid(depth_plot, col_plot, ncol = 1, align = "v", rel_heights = c(1,0.4))
combined_5c
```

```{r save_plot}
ggsave('fig_5C_combined_out.pdf', plot=combined_5c, width=4, height=4.5, units='in', dpi=300, device = cairo_pdf) #useDingbats = FALSE,
```
